docs/EVENT_BUS_GUIDE.md

```diff
@@ -0,0 +1,388 @@
+# Event Bus System Guide
+
+This guide explains how to use the new event bus system for managing asynchronous state updates in your application.
+
+## Overview
+
+The event bus system provides a centralized way to handle asynchronous operations and their state changes across your application. It solves the common problem of components not updating properly when async operations complete.
+
+## Key Benefits
+
+- **Decoupled Components**: Components don't need to know about each other's async operations
+- **Centralized Event Handling**: All async events flow through a single system
+- **Real-time Updates**: Components automatically update when relevant events occur
+- **Easy Debugging**: Event history makes it easy to trace async operation flows
+- **Type Safety**: Full TypeScript support with typed events
+
+## Core Concepts
+
+### Event Types
+
+The system defines several event types for different async operations:
+
+```typescript
+type AsyncEventType =
+  | "file.upload.started"
+  | "file.upload.progress"
+  | "file.upload.completed"
+  | "file.upload.failed"
+  | "autorag.sync.started"
+  | "autorag.sync.progress"
+  | "autorag.sync.completed"
+  | "autorag.sync.failed"
+  | "campaign.created"
+  | "campaign.updated"
+  | "campaign.deleted"
+  | "snippet.generated"
+  | "snippet.approved"
+  | "snippet.rejected";
+```
+
+### Event Structure
+
+All events follow a consistent structure:
+
+```typescript
+interface BaseEvent {
+  type: AsyncEventType;
+  timestamp: number;
+  source: string; // Component or service that emitted the event
+}
+```
+
+## Usage Examples
+
+### 1. Basic Event Emission
+
+```typescript
+import { useEventEmitter } from "../hooks/useAsyncState";
+
+function MyComponent() {
+  const emit = useEventEmitter();
```

Comment:
i now this i just the readme, but assuming it's accurate:

should really just be called `useEvent` and "emit" should be called `send`